frontend/javascripts/oxalis/view/action-bar/create_animation_modal.tsx

```diff
@@ -171,32 +176,34 @@ function CreateAnimationModal(props: Props) {
       (bb) => bb.id === selectedBoundingBoxId,
     )!.boundingBox;
 
-    // Submit currently visible pre-computed meshes
-    let meshSegmentIds: number[] = [];
-    let meshFileName: string | undefined;
-    let segmentationLayerName: string | undefined;
-
-    const visibleSegmentationLayer = Model.getVisibleSegmentationLayer();
-
-    if (visibleSegmentationLayer) {
-      const availableMeshes = state.localSegmentationData[visibleSegmentationLayer.name].meshes;
-      if (availableMeshes == null) {
-        throw new Error("There is no mesh data in localSegmentationData.");
-      }
-      meshSegmentIds = Object.values(availableMeshes as Record<number, MeshInformation>)
-        .filter((mesh) => mesh.isVisible && mesh.isPrecomputed)
-        .map((mesh) => mesh.segmentId);
-
-      const currentMeshFile =
-        state.localSegmentationData[visibleSegmentationLayer.name].currentMeshFile;
-      meshFileName = currentMeshFile?.meshFileName;
-
-      if (visibleSegmentationLayer.fallbackLayerInfo) {
-        segmentationLayerName = visibleSegmentationLayer.fallbackLayerInfo.name;
-      } else {
-        segmentationLayerName = visibleSegmentationLayer.name;
-      }
-    }
+    // Submit currently visible pre-computed & ad-hoc meshes
+    const axis = "";
+    const layerNames = Object.keys(state.localSegmentationData);
+    const { preferredQualityForMeshAdHocComputation } = state.temporaryConfiguration;
+
+    const meshes: RenderAnimationOptions["meshes"] = layerNames.flatMap((layerName) => {
+      const meshInfos = state.localSegmentationData[layerName]?.meshes?.[axis] || {};
+
+      return Object.values(meshInfos)
+        .filter((meshInfo: MeshInformation) => meshInfo.isVisible)
+        .flatMap((meshInfo: MeshInformation) => {
+          const layer = getLayerByName(state.dataset, layerName) as APISegmentationLayer;
+          const hasAFallbackLayer = hasFallbackLayer(layer);
+          const fullLayerName = layer.fallbackLayerInfo?.name || layerName;
+
+          const adhoc_mag_index = getResolutionInfo(layer.resolutions).getClosestExistingIndex(
+            preferredQualityForMeshAdHocComputation,
+          );
+          const adhoc_mag = layer.resolutions[adhoc_mag_index];
```

Comment:
I think, this can be moved above `Object.values(meshInfos)`, because the code only depends on the layerName, right?

Reply:
Done